e2e/deployment/deployment.go

```diff
@@ -34,29 +34,33 @@ func (tc *DeploymentTest) BeforeAll(f *framework.F) {
 func (tc *DeploymentTest) TestDeploymentAutoPromote(f *framework.F) {
 	t := f.T()
 	nomadClient := tc.Nomad()
+	run := structs.DeploymentStatusRunning
 	uuid := uuid.Generate()
+	// unique each run, cluster could have previous jobs
 	jobId := "deployment" + uuid[0:8]
 	tc.jobIds = append(tc.jobIds, jobId)
 	e2eutil.RegisterAndWaitForAllocs(t, nomadClient, "deployment/input/deployment_auto0.nomad", jobId)
+	deploy := e2eutil.DeploymentsForJob(nomadClient, jobId)[0]
```

Comment:
Should we protect against a nil or empty result here?  This would panic and makes it a bit harder to debug test.